users/signals.py

```diff
@@ -1,13 +1,13 @@
-from django.contrib.auth.models import User
 from django.db.models.signals import post_save
 from django.dispatch import receiver
 
+from users.models import CustomUser
 from .models import Profile, Role
 
 __all__ = ['create_profile', 'save_profile']
 
 
-@receiver(post_save, sender=User)
+@receiver(post_save, sender=CustomUser)
```

Comment:
Generally speaking it would be better to use the [get_user_model](https://docs.djangoproject.com/en/2.1/topics/auth/customizing/#django.contrib.auth.get_user_model) helper here instead of direct import